src/nn/add_into.rs

```diff
@@ -23,12 +26,19 @@ use super::{BuildModule, BuildOnDevice, Module, ModuleMut, ResetParams, ToDevice
 #[derive(Debug, Default, Clone)]
 pub struct AddInto<T>(pub T);
 
-impl<T: GradientUpdate<D, E>, D: Device<E>, E: Dtype> GradientUpdate<D, E> for AddInto<T> {
-    fn update<U>(&mut self, updater: &mut U, unused: &mut UnusedTensors) -> Result<(), <D>::Err>
-    where
-        U: ParamUpdater<D, E>,
-    {
-        self.0.update(updater, unused)
+impl<
+        const N: usize,
+        const M: usize,
+        T: VisitTensorGroups<N, M, E, D> + std::fmt::Debug,
+        E: Dtype,
+        D: DeviceStorage,
+    > VisitTensorGroups<N, M, E, D> for AddInto<T>
+{
+    fn visit_groups<F: TensorVisitor<N, M, E, D>>(
+        mut self_refs: ModuleGroup<N, M, Self>,
+        func: &mut F,
+    ) -> Result<(), F::Err> {
+        self_refs.map(|s| &s.0, |s| &mut s.0, "0.").visit(func)
     }
```

Comment:
Thoughts on including the visitor in the ModuleGroup object, and then call it inside .map?

If we can move the E/D generics to the call method, this shouldn't be an issue.

Something like:

```suggestion
    fn visit_groups<F: TensorVisitor<N, M>>(
        &mut self_refs: ModuleGroup<N, M, Self, F>,
    ) -> Result<(), F::Err> {
        self_refs.map(|s| &s.0, |s| &mut s.0, "0.")
    }
```

We could even go a step further and make ModuleGroup itself a TensorVisitor (wrapped around another TensorVisitor):

```suggestion
    fn visit_groups<V: TensorVisitor<N, M>>(
        &mut visitor: V,
    ) -> Result<(), F::Err> {
        visitor.map(|s| &s.0, |s| &mut s.0, "0.")
    }
```

Reply:
I've implemented your first suggestion and have renamed some things to make the new functionality of ModuleGroups make more sense. I haven't implemented your second suggestion because I don't really think I understand what you're getting at. What would the call implementation of ModuleGroup look like? How would we treat TensorVisitors not wrapped in a ModuleGroup?